c_api/docs/taichi/taichi_core.h.md

```diff
@@ -0,0 +1,410 @@
+---
+sidebar_position: 1
+---
+
+# Core Functionality
+
+Taichi Core exposes all necessary interfaces to offload AOT modules to Taichi. Here lists the features universally available disregards to any specific backend. These APIs are still in active development so is subject to change.
+
+## Availability
+
+Taichi C-API has bridged the following backends:
+
+|Backend|Offload Target|Maintenance Tier|
+|-|-|-|
+|Vulkan|GPU|Tier 1|
+|CUDA (LLVM)|GPU (NVIDIA)|Tier 1|
+|CPU (LLVM)|CPU|Tier 1|
+|DirectX 11|GPU (Windows)|N/A|
+|Metal|GPU (macOS, iOS)|N/A|
+|OpenGL|GPU|N/A|
+
+The backends with tier 1 support are the most intensively developed and tested ones. In contrast, you would expect a delay in fixes against minor issues on tier 2 backends. The backends currently unsupported might become supported. Among all the tier 1 backends, Vulkan has the most outstanding cross-platform compatibility, so most of the new features will be first available on Vulkan.
+
+For convenience, in the following text (and other C-API documentations), the term **host** refers to the user of the C-API; the term **device** refers to the logical (conceptual) compute device that Taichi Runtime offloads its compute tasks to. A *device* might not be an actual discrete processor away from the CPU and the *host* MAY NOT be able to access the memory allocated on the *device*.
+
+Unless explicitly explained, **device**, **backend**, **offload targer** and **GPU** are used interchangeably; **host**, **user code**, **user procedure** and **CPU** are used interchangeably too.
```

Comment:
```suggestion
Unless otherwise specified, **device**, **backend**, **offload targer**, and **GPU** are interchangeable; **host**, **user code**, **user procedure**, and **CPU** are interchangeable.
```